### PR TITLE
Fixes nades erroniously appearing primed when they aren't

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Yes. Tested by putting a nade into a GL and taking it out, usually this makes it appear like it's about to go off (with the flashing !)
## Why It's Good For The Game

Bug bad.
## Changelog
:cl:
fix: Fixed nades erroniously appearing primed when they aren't
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
the ! is specifically for dangerous nades, not all nades, afaik, so it check for both dangerous and active.

**Review by Lumipharon — APPROVED** [review]

File: code/game/objects/items/explosives/grenades/grenade.dm [hunk 1]
```diff
@@ -83,7 +83,7 @@
 
 /obj/item/explosive/grenade/update_overlays()
 	. = ..()
-	if(active)
+	if(active && dangerous)
 		overlays += new /obj/effect/overlay/danger
```
Comment (Lumipharon) [inline code/game/objects/items/explosives/grenades/grenade.dm:87]: Oh, this is also wrong, although I know its old code. update overlays should just return a list of overlays it should currently have. So:
. += new /obj/effect/overlay/danger

**Review by Lumipharon — CHANGES_REQUESTED** [review]
comment

**Review by Lumipharon — APPROVED** [review]

**Comment by Derrick2Electricboogaloo** [comment]
Finally!

**Pull request merged**